egui/src/widgets/plot/mod.rs

```diff
@@ -661,6 +741,29 @@ impl Plot {
         // --- Bound computation ---
         let mut bounds = *last_screen_transform.bounds();
 
+        // Find the cursors from other plots we need to draw
+        let draw_cursors: Vec<Cursor> = if let Some(group) = linked_cursors.as_ref() {
+            let mut frames = group.frames.borrow_mut();
+
+            // Look for our previous frame
+            let index = frames
+                .iter()
+                .enumerate()
+                .find(|(_, frame)| frame.id == plot_id)
+                .map(|(i, _)| i);
+
+            // Remove our previous frame and all older frames as these are no longer displayed.
+            index.map(|index| frames.drain(0..=index));
```

Comment:
I don't really follow this. What is `frames` sorted on? The docstring doesn't mention it being ordered by anything.

Reply:
`frames` is the oldest drawn widget to the newest. Each time a plot widget is drawn, it is appended to `frames`. Previous entries are removed here to avoid unbounded growth.

Reply:
Oh, now I see. Thanks for explaining! Could you please add those explanations to the code and then I will merge this!

Reply:
I added some more explanations.